src/query-builder/json-path-builder.ts

```diff
@@ -0,0 +1,276 @@
+import { AliasedExpression, Expression } from '../expression/expression.js'
+import { AliasNode } from '../operation-node/alias-node.js'
+import { IdentifierNode } from '../operation-node/identifier-node.js'
+import {
+  JSONPathLegNode,
+  JSONPathLegType,
+} from '../operation-node/json-path-leg-node.js'
+import { JSONPathNode } from '../operation-node/json-path-node.js'
+import { JSONPathReferenceNode } from '../operation-node/json-path-reference-node.js'
+import { isOperationNodeSource } from '../operation-node/operation-node-source.js'
+import { OperationNode } from '../operation-node/operation-node.js'
+import { RawNode } from '../operation-node/raw-node.js'
+import { ReferenceNode } from '../operation-node/reference-node.js'
+
+export class JSONPathBuilder<S, O = S> {
+  readonly #node: ReferenceNode | JSONPathNode
```

Comment:
Isn't this currently always a `ReferenceNode`? What's the use case for `JSONPathNode`?

`#createBuilderWithPathLeg` creates a new `ReferenceNode` if `#node` is a `ReferenceNode` and since it's a `ReferenceNode` initially --> it's always `ReferenceNode`.

Reply:
You're right.

The PR started off with just a `JSONPathBuilder` that could be used to generate `$.key[0]`, since this is a thing that can be used in many ways outside of `column->`, as a JSON function argument. Once I added the `column->` stuff, it became less important to support as a standalone, and eventually, I dropped the ability to use it that way for now - to focus the PR around `column->` which is way more common.